src/rpc/core_rpc_server.cpp

```diff
@@ -1522,6 +1522,53 @@ namespace cryptonote
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
+  bool core_rpc_server::on_get_limit(const COMMAND_RPC_GET_LIMIT::request& req, COMMAND_RPC_GET_LIMIT::response& res)
+  {
+    res.limit_down = epee::net_utils::connection_basic::get_rate_down_limit();
+    res.limit_up = epee::net_utils::connection_basic::get_rate_up_limit();
+    res.status = CORE_RPC_STATUS_OK;
+    return true;
+  }
+  //------------------------------------------------------------------------------------------------------------------------------
+  bool core_rpc_server::on_set_limit(const COMMAND_RPC_SET_LIMIT::request& req, COMMAND_RPC_SET_LIMIT::response& res)
+  {
+    // -1 = reset to default
+    //  0 = do not modify
+
+    if (req.limit_down > 0)
+    {
+      epee::net_utils::connection_basic::set_rate_down_limit(req.limit_down);
+    }
+    else if (req.limit_down < 0)
+    {
+      if (req.limit_down != 1)
+      {
+        res.status = CORE_RPC_ERROR_CODE_WRONG_PARAM;
+        return false;
+      }
+      epee::net_utils::connection_basic::set_rate_down_limit(nodetool::default_limit_down * 1024);
+    }
+
+    if (req.limit_up > 0)
+    {
+      epee::net_utils::connection_basic::set_rate_up_limit(req.limit_up);
+    }
+    else if (req.limit_up < 0)
+    {
+      if (req.limit_up != 1)
```

Comment:
That's odd, and I think that's not what you meant.
If also seems a bit complicated. I'd imagine something like:
uint64_t l = (req.limit_up >= 0)  limit_up = node_tool::default_limit_up ? req.limit_up;
if (l) epee::net_utils::set_rate_limit_up(l);

Reply:
That's part of the range check and is indeed what I mean.

Reply:
Woops it should actually be `-1` :smile: